src/formatting_stack/formatters/how_to_ns.clj

```diff
@@ -16,13 +19,33 @@
                              :import-square-brackets? false})
 
 (defn format! [{:keys [how-to-ns-options]} files]
-  (->> (remove #(str/ends-with? % ".edn") files)
+  (->> (strategies/exclude-edn :files files)
        (process-in-parallel! (fn [filename]
                                (how-to-ns.main/fix [filename] how-to-ns-options))))
   nil)
 
+(defn lint! [{:keys [how-to-ns-options]} files]
+  (->> (strategies/exclude-edn :files files)
+       (process-in-parallel! (fn [filename]
+                               (let [contents  (slurp filename)
+                                     formatted (how-to-ns/format-initial-ns-str contents how-to-ns-options)]
+                                 (when-not (= contents formatted)
+                                   (let [diff (diff/unified-diff filename contents formatted)]
+                                     (->> (diff->line-numbers diff)
+                                          (mapv (fn [{:keys [start]}]
+                                                  {:filename filename
+                                                   :diff diff
+                                                   :line start
+                                                   :column 0
+                                                   :level :warning
+                                                   :msg "Badly formatted namespace"
```

Comment:
Better: `Badly formatted ns form`, since `namespace` can also denote the whole file.

Also maybe I'd try to avoid `Badly` since it sounds judgemental. Using `:warning-details-url "https://stuartsierra.com/2016/clojure-how-to-ns.html"` can compensate

Reply:
Good call, went with `"Detected unsorted, renamed or extra clauses in the ns format"` and the url 👍